example/otlp/main.go

```diff
@@ -0,0 +1,99 @@
+// Example application showcasing opentelemetry Go using the OTLP wire
+// protocal
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"go.opentelemetry.io/otel/api/global"
+	"go.opentelemetry.io/otel/api/kv"
+	"go.opentelemetry.io/otel/api/metric"
+	"go.opentelemetry.io/otel/exporters/otlp"
+	"go.opentelemetry.io/otel/sdk/metric/controller/push"
+	"go.opentelemetry.io/otel/sdk/metric/selector/simple"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+)
+
+// Initializes an OTLP exporter, and configures the corresponding trace and
+// metric providers.
+func initProvider() (*otlp.Exporter, *push.Controller) {
+	exp, err := otlp.NewExporter(
+		otlp.WithInsecure(),
+		otlp.WithAddress("localhost:55680"),
+	)
+	handleErr(err, "Failed to create exporter: $v")
+
+	traceProvider, err := sdktrace.NewProvider(
+		sdktrace.WithConfig(sdktrace.Config{DefaultSampler: sdktrace.AlwaysSample()}),
+		sdktrace.WithSyncer(exp),
+	)
+	handleErr(err, "Failed to create trace provider: %v")
+
+	pusher := push.New(
+		simple.NewWithExactDistribution(),
+		exp,
+		push.WithStateful(true),
+		push.WithPeriod(2*time.Second),
+	)
+
+	global.SetTraceProvider(traceProvider)
+	global.SetMeterProvider(pusher.Provider())
+	pusher.Start()
+
+	return exp, pusher
+}
+
+func main() {
+	exp, pusher := initProvider()
+	defer exp.Stop()
+	defer pusher.Stop() // pushes any last exports to the receiver
+
+	tracer := global.Tracer("mage-sense")
+	meter := global.Meter("mage-read")
+
+	commonLabels := []kv.KeyValue{
```

Comment:
Can you add comments like "these labels describe a DD room. They represents Resource"?

Reply:
Done